### PR TITLE
Login screen redesign

### PR DESCRIPTION
**Pull request by KassFlute**
Completely redesign the login screen UI according to the Figma and personal decisions (no back-end was touched).
- Use Theme colors instead of hard-coded ones
- Tested on small phone emulator because a lot of different padding are used
- Rework all tests for the login screen and the according test tags

### CODE REVIEW

**Comment by KassFlute** [comment]
- Merged main into here to test with the new logo

**Comment by KassFlute** [comment]
100% line coverage 😎

File: app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt [hunk 1]
```diff
@@ -54,87 +69,197 @@ fun LoginScreen(authentication: Authentication, modifier: Modifier = Modifier) {
...
+          ) {
+            Text(
+                text = "Your neighbors are already here !",
+                style =
+                    MaterialTheme.typography.headlineMedium.copy(
+                        fontWeight = FontWeight.SemiBold, textAlign = TextAlign.Center),
+            )
+            val images =
+                listOf(
+                    R.drawable.multiprises,
```
Comment (Patrick-Pataky) [inline app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt:105]: Open Question: do we want to have actual images from the database or only "template" images ?
Personally, I think the app would look more active if dynamic images were used.

File: app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt [hunk 2]
```diff
@@ -54,87 +69,197 @@ fun LoginScreen(authentication: Authentication, modifier: Modifier = Modifier) {
...
+                  Modifier.fillMaxWidth()
+                      .height(120.dp)
+                      .background(color = MaterialTheme.colorScheme.background))
+        }
+    ClickableText(
+        modifier =
+            Modifier.align(Alignment.BottomCenter)
+                .padding(bottom = 70.dp)
+                .testTag("SeeMoreClickableText"),
+        text = AnnotatedString(text = "See 99+ more"),
```
Comment (Patrick-Pataky) [inline app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt:229]: Same for here, do we really want static text ?

Comment (Patrick-Pataky) [reply app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt:229]: Also the text seems to be black when dark mode is enabled, which is hard to read. The MaterialTheme is missing

File: app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt [hunk 3]
```diff
@@ -54,87 +69,197 @@ fun LoginScreen(authentication: Authentication, modifier: Modifier = Modifier) {
...
+                  modifier = Modifier.testTag("GoogleLoginButton"),
+                  name = "Sign in with Google",
+                  icon = R.drawable.google,
+                  onClick = {
+                    Log.w(TAG, "push button to sign in (Google)")
+                    authentication.signIn()
+                  })
+            }
+            item {
+              LoginOptionButton(
```
Comment (Patrick-Pataky) [inline app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt:158]: The way you modularized code is really appreciated, but is it worth it to add another login button if it is useless ? 
I suggest you that you remove the second button from here, and the day we decide to add Facebook, Tequila or Email authentication we create a new PR that adds it here. What do you think ?

File: app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt [hunk 4]
```diff
@@ -54,87 +69,197 @@ fun LoginScreen(authentication: Authentication, modifier: Modifier = Modifier) {
...
-              }
+/**
+ * Custom composable to display a grid of images blurred at the end.
+ *
+ * @param modifier Modifier to apply to this layout.
+ * @param images List of image resources to display.
+ * @param onShowSignInChange Callback to change the visibility of the sign in screen.
+ */
+@Composable
+fun ImageGrid(
```
Comment (Patrick-Pataky) [inline app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt:177]: The grid is really well-designed, nice job !

**Review by Patrick-Pataky — CHANGES_REQUESTED** [review]
Overall really nice improvements, it was time to change the old Login Screen !  
I think there are a few changes that could be discussed, but great job!

File: app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt (hunk 1 above)
Comment (Valentin889) [reply app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt:105]: @AntoineBut is working on that and it should be merge soon

File: app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt (hunk 2 above)
Comment (Valentin889) [reply app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt:229]: You want the text to by dynamic ? Except for language I don't really see the point, can you explain ?
Good catch for the materialTheme

**Review by Valentin889 — CHANGES_REQUESTED** [review]
This is very nice, I would just add something on the signin with an other method button so the user understand it's not available at the moment, right now nothing happens when you clic on it

File: app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt (hunk 1 above)
Comment (KassFlute) [reply app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt:105]: We can make a future task of replacing these sample images by real ones. But for now, as the images are not working and the db is not filled with real data, I think it's better to do it like this. And as I did it with a parameter image list, it should be easy to replace them.

File: app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt (hunk 2 above)
Comment (KassFlute) [reply app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt:229]: I guess it's about the number. But same argument as before, before we have real data in the db it's better like this.

Comment (KassFlute) [reply app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt:229]: I added a theme color and checked in dark mode. It seems good now.

File: app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt (hunk 3 above)
Comment (KassFlute) [reply app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt:158]: Yeah I forgot to remove the second login option, it was to test the layout with a second login option for when we will add one

**Comment by KassFlute** [comment]
> This is very nice, I would just add something on the signin with an other method button so the user understand it's not available at the moment, right now nothing happens when you clic on it

I removed it, it was just to test the layout with it for when we will add a second login option.

**Comment by KassFlute** [comment]
thanks for your reviews, please review again and I'll merge (after the CI)

File: app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt (hunk 1 above)
Comment (Valentin889) [reply app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt:105]: Agreed

File: app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt (hunk 2 above)
Comment (Valentin889) [reply app/src/main/java/com/android/partagix/ui/screens/LoginScreen.kt:222]: Nice fix

**Comment by Valentin889** [comment]
Seems good to me. If the CI agree then I think we can merge

**Review by Patrick-Pataky — APPROVED** [review]
Great job !

**Pull request merged**